libs/community/langchain_community/memory/motorhead_memory.py

```diff
@@ -0,0 +1,100 @@
+from typing import Any, Dict, List, Optional
+
+import requests
+from langchain_core.messages import get_buffer_string
+
+try:
+    # Temporarily tuck import in a conditional import until
+    # community pkg becomes dependent on langchain core
+    from langchain_core.memory.chat_memory import BaseChatMemory
```

Comment:
why the try/except if this is importing from core

Reply:
good catch it was langchain rather than langchain_core